src/util/createInstantSearch.js

```diff
@@ -0,0 +1,115 @@
+import instantsearch from 'instantsearch.js/es/index';
+import algoliaHelper from 'algoliasearch-helper';
+const { SearchParameters, SearchResults } = algoliaHelper;
+
+export const createInstantSearch = ({ searchClient, indexName, options }) => {
+  const search = instantsearch({
+    searchClient,
+    indexName,
+    ...options,
+  });
+
+  search._isSsr = true;
+
+  // main API for SSR, called in asyncData of a root component which contains instantsearch
+  search.findResultsState = params => {
+    search.helper = algoliaHelper(searchClient, indexName, {
+      // parameters set by default
+      highlightPreTag: '__ais-highlight__',
+      highlightPostTag: '__/ais-highlight__',
+      ...params,
```

Comment:
We can move the spread operator above the two other, otherwise the options can accidentally override the provided value.